packages/client/hmi-client/src/services/charts.ts

```diff
@@ -875,22 +875,44 @@ export function createFunmanStateChart(
 ) {
 	if (isEmpty(trajectories)) return null;
 
+	const MAX = 99999999999;
 	const globalFont = 'Figtree';
 
-	// Find min/max values to set an appropriate viewing range for y-axis
-	const minY = Math.floor(Math.min(...trajectories.map((d) => d.values[stateId])));
-	const maxY = Math.ceil(Math.max(...trajectories.map((d) => d.values[stateId])));
-
-	// Show checks for the selected state
 	const stateIdConstraints = constraints.filter((c) => c.variables.includes(stateId));
-	const modelChecks = stateIdConstraints.map((c) => ({
-		legendItem: FunmanChartLegend.ModelChecks,
-		startX: c.timepoints.lb,
-		endX: c.timepoints.ub,
+
+	// Find min/max values to set an appropriate viewing range for y-axis
+	const lowerBounds = stateIdConstraints.map((c) => c.additive_bounds.lb);
+	const upperBounds = stateIdConstraints.map((c) => c.additive_bounds.ub);
+	const yPoints = trajectories.map((t) => t.values[stateId]);
+
+	const potentialMinYs = focusOnModelChecks && !isEmpty(lowerBounds) ? lowerBounds : yPoints;
+	const potentialMaxYs = focusOnModelChecks && !isEmpty(upperBounds) ? upperBounds : yPoints;
+	let minY = Math.floor(Math.min(...potentialMinYs));
+	let maxY = Math.ceil(Math.max(...potentialMaxYs));
+	// Limit to MAX as very large numbers cause the chart to have NaN in the y domain
+	minY = minY < -MAX ? -MAX : minY;
+	maxY = maxY > MAX ? MAX : maxY;
```

Comment:
I think this needs some more investigation here, if it is the case that Funman's result goes to infinity, then that may indicate that something had gone wrong and we either need to communicate it to the Funman team or perhaps change the process logic on our side.

Do we have any additional data on this? We should only mask problems as a last resort.

Reply:
I don't get an `Infinity` but I do get a `1.7e308`. It renders fine if I have a y range of `[0, 1.7e308]` but it blows up when I have `[-1.7e308, 1.7e308]` giving me this:
![image](https://github.com/user-attachments/assets/3099b06a-b101-4ff0-a6df-4624cde573c0)

Reply:
Do you still have the request and response payloads? They would be useful to have, to determine what you are asking for, and what exactly you are getting back.

If the request looks sensible and the response does not, we should bump this over the Funman team to take a look, check in with @liunelson if you need a hand determining the sanity of either one.

In the meantime, we should probably add a FIXME here indicating that this is likely an abnormal event, and let's "console.error" some information here as well so it is more apparent.

Reply:
This happens when we check off the "Focus on model checks" box. Since the boxes are going to be drawn to infinity this makes sense. 

https://app.staging.terarium.ai/projects/34e7b955-f408-4e8b-be27-9429480ad6aa/workflow/ae644619-7fe5-40d7-bb90-880c447ba9a4?operator=330e2760-d3e1-4204-b8ed-08be949af5ed

https://github.com/user-attachments/assets/4cde8689-ce44-4ff3-9617-56f98bb0283a